command/helper/helper.go

```diff
@@ -531,3 +543,67 @@ func DirectoryExists(directoryPath string) bool {
 
 	return true
 }
+
+// PredeployStakingSC is a helper method for setting up the staking smart contract account,
+// using the passed in validators as prestaked validators
+func PredeployStakingSC(
+	premineMap map[types.Address]*chain.GenesisAccount,
+	validators []types.Address,
+) error {
+	// Set the code for the staking smart contract
+	// Code retrieved from https://github.com/0xPolygon/staking-contracts
+	scHex, _ := hex.DecodeHex(StakingSCBytecode)
+	stakingAccount := &chain.GenesisAccount{
+		Code: scHex,
+	}
+
+	// Parse the default staked balance value into *big.Int
+	val := DefaultStakedBalance
+	bigDefaultStakedBalance, err := types.ParseUint256orHex(&val)
+	if err != nil {
+		return fmt.Errorf("unable to generate DefaultStatkedBalance, %v", err)
+	}
+
+	// Generate the empty account storage map
+	storageMap := make(map[types.Hash]types.Hash)
+	bigTrueValue := big.NewInt(1)
+	stakedAmount := big.NewInt(0)
+	for indx, validator := range validators {
+		// Update the total staked amount
+		stakedAmount.Add(stakedAmount, bigDefaultStakedBalance)
+
+		// Get the storage indexes
+		storageIndexes := staking.GetStorageIndexes(validator, int64(indx))
+
+		// Set the value for the validators array
+		storageMap[types.BytesToHash(storageIndexes.ValidatorsIndex)] = types.BytesToHash(
+			validator.Bytes(),
+		)
+
+		// Set the value for the address -> validator array index mapping
+		storageMap[types.BytesToHash(storageIndexes.AddressToIsValidatorIndex)] = types.BytesToHash(bigTrueValue.Bytes())
+
+		// Set the value for the address -> staked amount mapping
+		storageMap[types.BytesToHash(storageIndexes.AddressToStakedAmountIndex)] = types.StringToHash(hex.EncodeBig(bigDefaultStakedBalance))
+
+		// Set the value for the address -> validator index mapping
+		storageMap[types.BytesToHash(storageIndexes.AddressToValidatorIndexIndex)] = types.StringToHash(hex.EncodeUint64(uint64(indx)))
+
+		// Set the value for the total staked amount
+		storageMap[types.BytesToHash(storageIndexes.StakedAmountIndex)] = types.BytesToHash(stakedAmount.Bytes())
+
+		// Set the value for the size of the validators array
+		storageMap[types.BytesToHash(storageIndexes.ValidatorsArraySizeIndex)] = types.StringToHash(hex.EncodeUint64(uint64(indx + 1)))
+	}
+
+	// Save the storage map
+	stakingAccount.Storage = storageMap
+
+	// Set the Staking SC balance to numValidators * defaultStakedBalance
+	stakingAccount.Balance = stakedAmount
+
+	// Add the account to the premine map so the executor can apply it to state
+	premineMap[types.StringToAddress("1001")] = stakingAccount
+
+	return nil
+}
```

Comment:
Shall we move this function into a specific package for PoS?

Reply:
Great idea 💯 

There's no need for this PoS logic to be bundled with other helper methods. We can move it to a separate package

Reply:
Fixed 👍 

7575d59